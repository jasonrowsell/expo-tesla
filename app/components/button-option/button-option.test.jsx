import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import ButtonOption from './button-option';

describe('<ButtonOption />', () => {
  test('has 1 child', () => {
    const onPress = jest.fn();
    const tree = renderer.create(<ButtonOption type="primary" content="content" onPress={onPress} />).toJSON();

    expect(tree.children.length).toBe(1);
  });

  test('renders correctly', () => {
    const onPress = jest.fn();
    const tree = render(<ButtonOption type="primary" content="content" onPress={onPress} />);

    expect(tree).toMatchSnapshot();
  });

  test('has the content', () => {
    const onPress = jest.fn();
    const { getByText } = render(<ButtonOption type="primary" content="content" onPress={onPress} />);

    getByText('content');
  });

  test('calls onPress function once clicked', () => {
    const onPress = jest.fn();
    const { getByTestId } = render(<ButtonOption type="primary" content="content" onPress={onPress} />);

    const button = getByTestId('button');
    fireEvent.press(button);

    expect(onPress).toHaveBeenCalled();
  });

  describe('primary type of button', () => {
    test('has backgroundColor of light gray', () => {
      const onPress = jest.fn();
      const { getByTestId } = render(<ButtonOption type="primary" content="content" onPress={onPress} />);

      const button = getByTestId('button');

      expect(button).toHaveStyle({
        backgroundColor: '#171A20CC',
      });
    });

    test('has font color of black', () => {
      const onPress = jest.fn();
      const { getByText } = render(<ButtonOption type="primary" content="content" onPress={onPress} />);

      const button = getByText('content');

      expect(button).toHaveStyle({
        color: '#FFFFFF',
      });
    });
  });

  describe('secondary type of button', () => {
    test('has backgroundColor of dark gray', () => {
      const onPress = jest.fn();
      const { getByTestId } = render(<ButtonOption type="secondary" content="content" onPress={onPress} />);

      const button = getByTestId('button');

      expect(button).toHaveStyle({
        backgroundColor: '#FFFFFFA6',
      });
    });

    test('has font color of light gray', () => {
      const onPress = jest.fn();
      const { getByText } = render(<ButtonOption type="secondary" content="content" onPress={onPress} />);

      const button = getByText('content');

      expect(button).toHaveStyle({
        color: '#171A20',
      });
    });
  });
});
