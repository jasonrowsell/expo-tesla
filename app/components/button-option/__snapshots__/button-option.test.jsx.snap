// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonOption /> renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#171A20CC",
        },
      ]
    }
    testID="button"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "500",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      content
    </Text>
  </View>
</View>
`;
